s3/src/main/resources/reference.conf

```diff
@@ -73,4 +73,7 @@ akka.stream.alpakka.s3 {
   # When using virtual hosted–style buckets with SSL, the S3 wild card certificate only matches buckets that do not contain periods.
   # Buckets containing periods will lead to certificate errors. In those cases it's useful to enable path-style access.
   path-style-access = false
+
+  # Custom endpoint url, used for alternate s3 implementations
+  endpoint-url = null
```

Comment:
I wonder if this is discouraged

> If you do need a setting to be optional: checking hasPath() in advance should be the same amount of code (in Java) as checking for null afterward, without the risk of NullPointerException when you forget. In Scala, you could write an enrichment class like this to use the idiomatic Option syntax:
https://github.com/lightbend/config#how-to-handle-defaults

Reply:
Perhaps? I guess the the alternate solution would be to include the key, commented out. It seems like it should be documented in the reference.conf file in some capacity or another.